qa/packaging/src/test/java/org/elasticsearch/packaging/test/MemoryLockingTests.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.packaging.test;
+
+import org.elasticsearch.packaging.util.ServerUtils;
+import org.elasticsearch.packaging.util.docker.DockerRun;
+
+import java.util.Map;
+
+import static org.elasticsearch.packaging.util.docker.Docker.runContainer;
+import static org.elasticsearch.packaging.util.docker.DockerRun.builder;
+
+public class MemoryLockingTests extends PackagingTestCase {
+
+    public void test10Install() throws Exception {
+        install();
+    }
+
+    public void test20MemoryLockingEnabled() throws Exception {
+        configureAndRun(Map.of(
+            "bootstrap.memory_lock",
+            "true",
+            "xpack.security.enabled",
+            "false",
+            "xpack.security.http.ssl.enabled",
+            "false",
+            "xpack.security.enrollment.enabled",
+            "false",
+            "discovery.type",
+            "single-node"));
+        // TODO: very locking worked. logs? check memory of process? at least we know the process started successfully
```

Comment:
I think the best (but complicated) way of doing this it to check the memory of the process, by calling VirtualQueryEx (and/or the equivalent function under linux/macos) from the test. 
It does work cross process, so it should be possible to do that even in a test that spawns an external cluster. No idea how to do that with a docker test though...
Another idea: expose the information we need via an `_nodes/stats` (under os.mem maybe?). Could be useful regardless (e.g. for diagnostics)